### PR TITLE
Add processor architecture to RUMDeviceInfo

### PR DESCRIPTION
**Pull request by fuzzybinary**
### What and why?

Flutter / Dart needs architecture in order to select the proper symbol file to process.  Currently, only add architecture if UIKit is imported

### How?

Add architecture when creating the DeviceInfo. I've added a few checks on the Integration tests to make sure that the architecture is present and is an expected value.

### Review checklist
- [x] Feature or bugfix MUST have appropriate tests (unit, integration)
- [x] Make sure each commit and the PR mention the Issue number or JIRA reference
- [ ] Add CHANGELOG entry for user facing changes

### Custom CI job configuration (optional)
- [x] Run unit tests
- [x] Run integration tests
- [x] Run smoke tests

### CODE REVIEW

File: Sources/Datadog/DatadogInternal/Context/DeviceInfo.swift [hunk 1]
```diff
@@ -69,13 +77,19 @@ extension DeviceInfo {
         processInfo: ProcessInfo = .processInfo,
         device: UIDevice = .current
     ) {
+        var architecture = "unknown"
+        if let archInfo = NXGetLocalArchInfo()?.pointee {
+            architecture = String(utf8String: archInfo.name) ?? "unknown"
+        }
```
Comment (ncreated) [inline Sources/Datadog/DatadogInternal/Context/DeviceInfo.swift:83]: I wonder if these problems do still exist (these are dated back to 2020):
* https://github.com/getsentry/sentry-cocoa/issues/635
* https://github.com/firebase/firebase-ios-sdk/issues/6128

TL;DR; This was causing app rejection in App Store submission. Any thoughts @fuzzybinary @maxep ? Is there any alternative way to get this information?

File: Tests/DatadogIntegrationTests/Scenarios/RUM/RUMManualInstrumentationScenarioTests.swift [hunk 2]
```diff
@@ -58,6 +58,15 @@ class RUMManualInstrumentationScenarioTests: IntegrationTests, RUMCommonAsserts
         let view1 = session.viewVisits[0]
         XCTAssertEqual(view1.name, "SendRUMFixture1View")
         XCTAssertEqual(view1.path, "Example.SendRUMFixture1ViewController")
+        XCTAssertNotNil(view1.viewEvents.last?.device)
+        XCTAssertNotNil(view1.viewEvents.last?.device?.architecture)
+        if let architecture = view1.viewEvents.last?.device?.architecture {
+            // i486 is the architecture of the VMs on bitrise
```
Comment (ncreated) [inline Tests/DatadogIntegrationTests/Scenarios/RUM/RUMManualInstrumentationScenarioTests.swift:64]: 🥴👍

**Review by ncreated — COMMENTED** [review]
Looks very good 👌. I left feedback on potential use of private API - it needs some more thoughts ⚠️.

File: Sources/Datadog/DatadogInternal/Context/DeviceInfo.swift (hunk 1 above)
Comment (fuzzybinary) [reply Sources/Datadog/DatadogInternal/Context/DeviceInfo.swift:83]: I actually saw those, but if I read things right, it looks like this only affected the Mac app store / Catalyst apps (which we don't support at the moment anyway?)

But even according to the Firebase issue, they didn't actually remove the call and haven't gotten any reports of being rejected since September.

Comment (maxep) [reply Sources/Datadog/DatadogInternal/Context/DeviceInfo.swift:83]: :/ this worries me a bit, but it looks like Firebase is still using this API without any report. So we can try with dogfooding and see if we got a rejection from the app store, WDYT? We would surely make sure the app team is ok with it!
My concern is that we aim to support multiplatform, so this would add debt for macos. We are not there yet and I prefer unblocking flutter and tackle that later 👍

Comment (fuzzybinary) [reply Sources/Datadog/DatadogInternal/Context/DeviceInfo.swift:83]: My guess is that API was flagged as private by accident.

**Review by maxep — APPROVED** [review]
👍
